### PR TITLE
Rest api V2 - add GET environments/{envId}/apis

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

https://gravitee.atlassian.net/browse/APIM-1162

## Description

Add get apis endpoint to list api (V2 & V4). The api list are fetch from the db directly and not use lucene. 

The next pr is for search apis with params and use lucene.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-jyyygrvgox.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v4/gravitee-apim-rest-api-management-v4-rest/src/main/resources/management-openapi-v4.yaml [hunk 1]
```diff
@@ -1123,6 +1124,88 @@ components:
             enum:
                 - BEST_MATCH
                 - DEFAULT
+        GenericApi:
+            type: object
+            title: GenericApi
+            properties:
```
Comment (ThibaudAV) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v4/gravitee-apim-rest-api-management-v4-rest/src/main/resources/management-openapi-v4.yaml]: not all props from GenericApiEntity but a good simple start. 
we can add other like categories, workflowState in dedicated PR if needed later

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/search/lucene/searcher/AbstractDocumentSearcher.java [hunk 2]
```diff
@@ -133,7 +133,7 @@ protected Optional<Query> buildFilterQuery(String apiReferenceField, Map<String,
                         filterApisQuery.add(new TermQuery(new Term(apiReferenceField, (String) value)), BooleanClause.Occur.SHOULD)
                     );
                 if (valuesAsCollection.size() > 0) {
-                    filtersQuery.add(filterApisQuery.build(), BooleanClause.Occur.SHOULD);
+                    filtersQuery.add(filterApisQuery.build(), BooleanClause.Occur.MUST);
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/search/lucene/searcher/AbstractDocumentSearcher.java]: This change is related to the `_search` endpoint. Maybe move it to the other PR?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/v4/impl/ApiServiceImpl_findAllTest.java [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+        );
+
+        assertThat(apis).isNotNull();
+        assertThat(apis.getContent().size()).isEqualTo(1);
+        assertThat(apis.getTotalElements()).isEqualTo(1);
+        assertThat(apis.getPageNumber()).isEqualTo(1);
+        assertThat(apis.getPageElements()).isEqualTo(1);
+
+        verify(apiRepository, times(1)).search(any(), any(), any(), any());
+    }
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/v4/impl/ApiServiceImpl_findAllTest.java:380]: Add test if user is not admin and they have no APIs to read, the repository is never called. (I can add this test if you'd like)

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v4/gravitee-apim-rest-api-management-v4-rest/src/main/java/io/gravitee/rest/api/management/v4/rest/mapper/ApiMapper.java [hunk 4]
```diff
@@ -25,18 +25,27 @@
...
 import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
 import org.mapstruct.factory.Mappers;
 
 @Mapper
 public interface ApiMapper {
     ApiMapper INSTANCE = Mappers.getMapper(ApiMapper.class);
 
-    Api convert(ApiEntity apiEntity);
+    void convert(ApiEntity apiEntity, @MappingTarget Api api);
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v4/gravitee-apim-rest-api-management-v4-rest/src/main/java/io/gravitee/rest/api/management/v4/rest/mapper/ApiMapper.java]: For this, we are missing the `default` method that we added for the args constructor. So either keep it as `Api convert(ApiEntity apiEntity);` or add in the default method.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/v4/api/GenericApiEntity.java [hunk 5]
```diff
@@ -20,6 +20,7 @@
 import io.gravitee.definition.model.DefinitionVersion;
 import io.gravitee.rest.api.model.PrimaryOwnerEntity;
 import io.gravitee.rest.api.model.Visibility;
+import io.gravitee.rest.api.model.WorkflowState;
```
Comment (ThibaudAV) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/v4/api/GenericApiEntity.java]: 🚮

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v4/gravitee-apim-rest-api-management-v4-rest/src/main/resources/management-openapi-v4.yaml [hunk 6]
```diff
@@ -94,7 +94,8 @@ paths:
                 - APIs
             summary: List APIs
             description: |-
-                # FIXME add a description
+                Get the list of generic APIs for a specific environment.
```
Comment (jourdiw) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v4/gravitee-apim-rest-api-management-v4-rest/src/main/resources/management-openapi-v4.yaml]: Using keyword "generic APIs" here but data is a list of Apis. Either leave out the "generic" adjective or better describe Api in its description or name to indicate that it's a generic API.

**Review by jourdiw — APPROVED** [review]

**Review by phiz71 — APPROVED** [review]

**Pull request merged**